### PR TITLE
Feature/output transform fn

### LINKED ISSUE #64: Add Filter2D infr to lib_nn
Add all associated Classes and Tests to support Filter2D kernels

### PR DESCRIPTION
**Pull request by andrewstanfordjason**
This PR adds int8 Output transforms. It requires the asm constants for the xcore implementation, specifically they are for the symmetric rounding work around.

### CODE REVIEW

File: lib_nn/api/OutputTransformFn.hpp [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+    int16_t *biases;         //[output_slice_channel_count];
+    int16_t *multipliers;    //[output_slice_channel_count];
+    int16_t *accu_modifier;  //[output_slice_channel_count];
+
+    Params(int32_t output_slice_channel_count,
+           OutputTransformValuesClamping *otv, int16_t *biases,
+           int16_t *multipliers, int16_t *accu_modifier);
+  };
+
+  Params *params;
```
Comment (ghost) [inline lib_nn/api/OutputTransformFn.hpp:231]: We are using private in the previous class, but not here. Private is also not being used in the following classes. Could you please take a look through all the classes and put all the relevant sections in private only exposing what is necessary?

File: lib_nn/src/cpp/OutputTransformFn.cpp [hunk 2]
```diff
@@ -0,0 +1,620 @@
...
+
+  auto vpu = nn::VPU();
+  vpu_vector_t vec_tmp;
+  uint32_t tmp;
+
+  vpu.vldr(vpu_vect_0x80);
+  vpu.vstrpv(output, write_mask);
+
+  vpu.vldd(acc->vD);
+  vpu.vldr(acc->vR);
```
Comment (ghost) [inline lib_nn/src/cpp/OutputTransformFn.cpp:592]: Should these functions be similar to the ones above? Meaning VLDR() VLDD() used above?

**Review by ghost — APPROVED** [review]
+1

File: lib_nn/src/cpp/OutputTransformFn.cpp (hunk 2 above)
Comment (andrewstanfordjason) [reply lib_nn/src/cpp/OutputTransformFn.cpp:592]: they are ok!

**Pull request merged**